java/impl/aws/src/main/java/co/worklytics/psoxy/AwsApiDataModeHybridHandler.java

```diff
@@ -0,0 +1,133 @@
+import co.worklytics.psoxy.aws.request.APIGatewayV1ProxyEventRequestAdapter;
+import co.worklytics.psoxy.aws.request.APIGatewayV2HTTPEventRequestAdapter;
+import co.worklytics.psoxy.aws.request.LambdaEventUtils;
+import co.worklytics.psoxy.gateway.HttpEventRequest;
+import co.worklytics.psoxy.gateway.HttpEventResponse;
+import co.worklytics.psoxy.gateway.impl.ApiDataRequestHandler;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * AWS lambda entrypoint that can handle API data requests via any of:
+ *   - cloud function URL invocations
+ *   - API Gateway events
+ *   - SQS events ( asynchronous processing of API data requests, that originally came from one of the above)
+ */
+@Log
+public class AwsApiDataModeHybridHandler implements RequestStreamHandler {
+
+    /**
+     * Static initialization allows reuse in containers
+     * {@link "https://aws.amazon.com/premiumsupport/knowledge-center/lambda-improve-java-function-performance/"}
+     */
+    static AwsContainer awsContainer;
+
+    // for parsing payloads that our code-serialized to JSON (Eg, SQS message content)
+    static ObjectMapper payloadMapper;
+    static ApiDataRequestHandler requestHandler;
+    static ResponseCompressionHandler responseCompressionHandler;
+    static LambdaEventUtils lambdaEventUtils;
+
+    static {
+        staticInit();
+    }
+
+    private static void staticInit() {
+        awsContainer = DaggerAwsContainer.create();
+        payloadMapper = awsContainer.objectMapper();
+        requestHandler = awsContainer.apiDataRequestHandler();
+        responseCompressionHandler = new ResponseCompressionHandler();
+        lambdaEventUtils = awsContainer.lambdaEventUtils();
+    }
+
+    @Override
+    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
+        // Read the full input stream into a tree
+        JsonNode rootNode = lambdaEventUtils.read(input);
+
+        ApiDataRequestHandler.ProcessingContext processingContext;
+
+        boolean base64Encoded = false;
+
+        // whatever the event is, convert to HttpEventRequest and invoke the request handler
+        if (lambdaEventUtils.isSQSEvent(rootNode)) {
+
+            JsonNode records = rootNode.get("Records");
+            if (records == null || records.isEmpty()) {
+                throw new IllegalArgumentException("SQS event has no records");
+            }
+            if (records.size() > 1) {
+                // well, tbh - why have this limitation? we certainly could batch in future if there's a use case
```

Comment:
agree

Reply:
that was initially specific to the webhook case; this handler is for the API data connector case. 

but I've refactored to handle arbitrary number of records.